addons/web/static/src/js/core/class_patcher.js

```diff
@@ -0,0 +1,48 @@
+odoo.define("web.class_patcher", function () {
+    "use strict";
+
+    function makePatchable(OriginalClass) {
+        let unpatchList = [];
+        class PatchableClass extends OriginalClass {}
+
+        PatchableClass.patch = function (name, patch) {
+            if (unpatchList.find(x => x.name === name)) {
+                throw new Error(`Class ${OriginalClass.name} already has a patch ${name}`);
+            }
+            if (!Object.prototype.hasOwnProperty.call(this, 'patch')) {
```

Comment:
what is the difference between this and simply `if (!this.hasOwnProperty('patch') {` ??

Reply:
https://eslint.org/docs/rules/no-prototype-builtins

Reply:
very interesting